src/process.cpp

```diff
@@ -18,248 +20,275 @@ void ManifoldPreprocess(Params &params, Model &m, ofstream &of)
-                               int(ch2.triangles[i][1] + ch1.points.size()), int(ch2.triangles[i][2] + ch1.points.size())});
-  merge.ComputeCH(ch);
-  SyncNorm(ch1, ch);
+    Model merge;
+    merge.points.insert(merge.points.end(), ch1.points.begin(), ch1.points.end());
+    merge.points.insert(merge.points.end(), ch2.points.begin(), ch2.points.end());
+    merge.triangles.insert(merge.triangles.end(), ch1.triangles.begin(), ch1.triangles.end());
+    for (int i = 0; i < (int)ch2.triangles.size(); i++)
+        merge.triangles.push_back({ int(ch2.triangles[i][0] + ch1.points.size()),
+                                   int(ch2.triangles[i][1] + ch1.points.size()), int(ch2.triangles[i][2] + ch1.points.size()) });
+    merge.ComputeCH(ch);
+    SyncNorm(ch1, ch);
 }
 
-double MergeConvexHulls(Model &m, vector<Model> &meshs, vector<Model> &cvxs, Params &params, ofstream &of, double epsilon, double threshold)
+double MergeConvexHulls(Model& m, vector<Model>& meshs, vector<Model>& cvxs, Params& params, ofstream& of, double epsilon, double threshold)
 {
-  cout << " - Merge Convex Hulls" << endl;
-  size_t nConvexHulls = (size_t)cvxs.size();
-  double h = 0;
+    cout << " - Merge Convex Hulls" << endl;
+    size_t nConvexHulls = (size_t)cvxs.size();
+    double h = 0;
 
-  if (nConvexHulls > 1)
-  {
-    // Populate the cost matrix
-    size_t idx = 0;
-    vector<double> costMatrix, precostMatrix;
-    costMatrix.resize(((nConvexHulls * nConvexHulls) - nConvexHulls) >> 1);    // only keeps the top half of the matrix
-    precostMatrix.resize(((nConvexHulls * nConvexHulls) - nConvexHulls) >> 1); // only keeps the top half of the matrix
-    int tmp = 0;
-    for (size_t p1 = 1; p1 < nConvexHulls; ++p1) // not calculate with itself, so start from 1
+    if (nConvexHulls > 1)
     {
-      for (size_t p2 = 0; p2 < p1; ++p2)
-      {
-        double dist = MeshDist(cvxs[p1], cvxs[p2]);
-        if (dist < threshold)
-        {
-          tmp++;
-          Model combinedCH;
-          MergeCH(cvxs[p1], cvxs[p2], combinedCH);
+        int bound = ((((nConvexHulls - 1) * nConvexHulls)) >> 1);
+        // Populate the cost matrix
+        size_t idx = 0;
+        vector<double> costMatrix, precostMatrix;
+        costMatrix.resize(bound);    // only keeps the top half of the matrix
+        precostMatrix.resize(bound); // only keeps the top half of the matrix
 
-          costMatrix[idx] = ComputeHCost(cvxs[p1], cvxs[p2], combinedCH, params.rv_k, params.resolution, params.seed);
-          precostMatrix[idx++] = max(ComputeHCost(meshs[p1], cvxs[p1], params.rv_k, 3000, params.seed),
-                                     ComputeHCost(meshs[p2], cvxs[p2], params.rv_k, 3000, params.seed));
-        }
-        else
-        {
-          costMatrix[idx++] = INF;
+        size_t p1, p2;
+#ifdef _OPENMP
+#pragma omp parallel for default(none) shared(costMatrix,precostMatrix,cvxs,params) private(p1,p2)

```

Comment:
```suggestion
#pragma omp parallel for default(none) shared(costMatrix,precostMatrix,cvxs,params,bound,threshold,meshs) private(p1,p2)
```
Make it work on Linux.